base/download.jl

```diff
@@ -35,9 +35,21 @@ function find_curl()
     end
 end
 
+# Clear libpath for invoking system executables that we generally don't want to be confused
+# by things like a libcurl that sits in Julia's private libdir.
+function clear_libpath(f::Function)
+    withenv(f,
+        "LD_LIBRARY_PATH" => nothing,
+        "DYLD_FALLBACK_LIBRARY_PATH" => nothing,
+        "DYLD_LIBRARY_PATH" => nothing,
+    )
```

Comment:
This is not thread-safe, right? https://github.com/JuliaLang/julia/issues/34726

Why not use `setenv`?

Reply:
You're right, that's just better.  Thanks!